### PR TITLE
fix leader change unstable UT

### PR DESCRIPTION
**Pull request by panda-sheep**
fix  https://github.com/vesoft-inc/nebula/pull/2012/checks?check_run_id=547541664
and other similar UT.

### CODE REVIEW

File: src/storage/test/AddEdgesTest.cpp [hunk 1]
```diff
@@ -17,7 +17,10 @@ namespace storage {
 
 TEST(AddEdgesTest, SimpleTest) {
     fs::TempDir rootPath("/tmp/AddEdgesTest.XXXXXX");
-    std::unique_ptr<kvstore::KVStore> kv = TestUtils::initKV(rootPath.path());
+    constexpr int32_t partitions = 6;
+    std::unique_ptr<kvstore::KVStore> kv = TestUtils::initKV(rootPath.path(), partitions,
+        {0, network::NetworkUtils::getAvailablePort()});
+    TestUtils::waitUntilAllElected(kv.get(), 0, {0, 1, 2, 3, 4, 5}/*partitions*/);
```
Comment (dangleptr) [inline src/storage/test/AddEdgesTest.cpp:23]: I suggest you move the waitUntilAllElected into init::KV() method.

Comment (panda-sheep) [reply src/storage/test/AddEdgesTest.cpp:23]: Good point， 👍

File: src/storage/test/TestUtils.h [hunk 2]
```diff
@@ -74,7 +74,31 @@ class TestUtils {
                                                             localhost,
                                                             workers);
         store->init();
-        sleep(1);
+
+        // When useMetaServer is ture we deal with it using waitUntilAllElected
```
Comment (Shylock-Hg) [inline src/storage/test/TestUtils.h:78]: Typo `true`.

Comment (panda-sheep) [reply src/storage/test/TestUtils.h:78]: Bright eyes,  👍

File: src/storage/test/TestUtils.h [hunk 3]
```diff
@@ -74,7 +74,31 @@ class TestUtils {
...
+                        }
+                    }
+                }
+                if (readyNum == partitionNumber) {
+                    LOG(INFO) << "All leaders have been elected!";
+                    break;
+                }
+                usleep(100000);
+            }
+        }
```
Comment (bright-starry-sky) [inline src/storage/test/TestUtils.h:101]: Here , Why not call waitUntilAllElected direct?

File: src/storage/test/AddVerticesTest.cpp [hunk 4]
```diff
@@ -17,7 +17,9 @@ namespace storage {
 
 TEST(AddVerticesTest, SimpleTest) {
     fs::TempDir rootPath("/tmp/AddVerticesTest.XXXXXX");
-    std::unique_ptr<kvstore::KVStore> kv = TestUtils::initKV(rootPath.path());
+    constexpr int32_t partitions = 6;
+    std::unique_ptr<kvstore::KVStore> kv = TestUtils::initKV(rootPath.path(), partitions,
+        {0, network::NetworkUtils::getAvailablePort()});
```
Comment (bright-starry-sky) [inline src/storage/test/AddVerticesTest.cpp:22]: You can avoid changing the code if there are parameter defaults. right ？

File: src/storage/test/TestUtils.h (hunk 3 above)
Comment (panda-sheep) [reply src/storage/test/TestUtils.h:101]: when  useMetaServer is true,  use ` waitUntilAllElected(kvstore::KVStore* kvstore, GraphSpaceID spaceId, int partNum)`
when   useMetaServer  is flase, use ```
 waitUntilAllElected(kvstore::KVStore* kvstore, GraphSpaceID spaceId,
        std::set<PartitionID> parts)

The difference between these two functions is whether to use metaserver, so the start of partId is also different, one starts from 0 and the other starts from 1.

```

File: src/storage/test/AddEdgesTest.cpp [hunk 5]
```diff
@@ -17,7 +17,9 @@ namespace storage {
 
 TEST(AddEdgesTest, SimpleTest) {
     fs::TempDir rootPath("/tmp/AddEdgesTest.XXXXXX");
-    std::unique_ptr<kvstore::KVStore> kv = TestUtils::initKV(rootPath.path());
+    constexpr int32_t partitions = 6;
+    std::unique_ptr<kvstore::KVStore> kv = TestUtils::initKV(rootPath.path(), partitions,
+        {0, network::NetworkUtils::getAvailablePort()});
```
[the remark this answers is not in the document]
Comment (panda-sheep) [reply src/storage/test/AddEdgesTest.cpp]: Good point, fixed.

File: src/storage/test/AddVerticesTest.cpp (hunk 4 above)
Comment (panda-sheep) [reply src/storage/test/AddVerticesTest.cpp]: 👍

File: src/storage/test/TestUtils.h [hunk 6]
```diff
@@ -74,7 +74,31 @@ class TestUtils {
                                                             localhost,
                                                             workers);
         store->init();
-        sleep(1);
+
+        // When useMetaServer is true, we deal with it using waitUntilAllElected.
+        if (useMetaServer) {
+            sleep(1);
```
Comment (Shylock-Hg) [inline src/storage/test/TestUtils.h:80]: `sleep(1)` here?

Comment (panda-sheep) [reply src/storage/test/TestUtils.h:80]: Yeah, when useMetaServer is true, follow the original logic

Comment (Shylock-Hg) [reply src/storage/test/TestUtils.h:80]: Yes, I think it's useless.

Comment (panda-sheep) [reply src/storage/test/TestUtils.h:80]: I think  here `sleep`, and the  following `waitUntilAllElected `  basically does not need to `sleep`  anymore.

Comment (panda-sheep) [reply src/storage/test/TestUtils.h:80]: This pr is not needed.

**Pull request merged**